neuralop/layers/spectral_convolution.py

```diff
@@ -268,24 +268,22 @@ def __init__(
         self.out_channels = out_channels
         self.joint_factorization = joint_factorization
 
-        # We index quadrands only
         # n_modes is the total number of modes kept along each dimension
-        # half_n_modes is half of that except in the last mode, correponding to
-        # the number of modes to keep in *each* quadrant for each dim
-        if isinstance(n_modes, int):
+        if isinstance(n_modes, int): # Should happen for 1D FNO only
             n_modes = [n_modes]
+        else:
+            n_modes = list(n_modes)
+        # The last mode has a redundacy as we use real FFT
+        # As a design choice we do the operation here to avoid users dealing with the +1
```

Comment:
I don't think this is documented in the class docstring, though that's where users would readily see it to know about this design. Consider adding this in the docstring.

Reply:
Yeah I want to polish the PR a little more